subxt/src/backend/mod.rs

```diff
@@ -0,0 +1,351 @@
+        Self {
+            hash,
+            _pointer: Some(Arc::new(inner)),
+        }
+    }
+
+    /// Return the hash of the referenced block.
+    pub fn hash(&self) -> H
+    where
+        H: Copy,
+    {
+        self.hash
+    }
+}
+
+/// A trait that a [`Backend`] can implement to know when some block
+/// can be unpinned: when this is dropped, there are no remaining references
+/// to the block that it's associated with.
+pub trait BlockRefT: Send + Sync + 'static {}
+
+/// A stream of some item.
+pub struct StreamOf<T>(Pin<Box<dyn Stream<Item = T> + Send + 'static>>);
+
+impl<T> Stream for StreamOf<T> {
+    type Item = T;
+    fn poll_next(
+        mut self: std::pin::Pin<&mut Self>,
+        cx: &mut std::task::Context<'_>,
+    ) -> std::task::Poll<Option<Self::Item>> {
+        self.0.poll_next_unpin(cx)
+    }
+}
+
+impl<T> std::fmt::Debug for StreamOf<T> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_tuple("StreamOf").field(&"<stream>").finish()
+    }
+}
+
+impl<T> StreamOf<T> {
+    /// Construct a new stream.
+    pub fn new(inner: Pin<Box<dyn Stream<Item = T> + Send + 'static>>) -> Self {
+        StreamOf(inner)
+    }
+
+    /// Returns the next item in the stream. This is just a wrapper around
+    /// [`StreamExt::next()`] so that you can avoid the extra import.
+    pub async fn next(&mut self) -> Option<T> {
+        StreamExt::next(self).await
+    }
+}
+
+/// A stream of [`Result<Item, Error>`].
+pub type StreamOfResults<T> = StreamOf<Result<T, Error>>;
+
+/// Runtime version information needed to submit transactions.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct RuntimeVersion {
+    /// Version of the runtime specification. A full-node will not attempt to use its native
+    /// runtime in substitute for the on-chain Wasm runtime unless all of `spec_name`,
+    /// `spec_version` and `authoring_version` are the same between Wasm and native.
+    pub spec_version: u32,
+
+    /// All existing dispatches are fully compatible when this number doesn't change. If this
+    /// number changes, then `spec_version` must change, also.
+    ///
+    /// This number must change when an existing dispatchable (module ID, dispatch ID) is changed,
+    /// either through an alteration in its user-level semantics, a parameter
+    /// added/removed/changed, a dispatchable being removed, a module being removed, or a
+    /// dispatchable/module changing its index.
+    ///
+    /// It need *not* change when a new module is added or when a dispatchable is added.
+    pub transaction_version: u32,
+}
+
+/// The status of the transaction.
+///
+/// If the status is [`TransactionStatus::InFinalizedBlock`], [`TransactionStatus::Error`],
+/// [`TransactionStatus::Invalid`] or [`TransactionStatus::Dropped`], then no future
+/// events will be emitted.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum TransactionStatus<Hash> {
+    /// Transaction is part of the future queue.
```

Comment:
I like the new `TransactionStatus`!

Reply:
It's basically much more similar to the one that the new APIs will return, so making this change now means that hopefully we won't need to change anything else as we implement the new backend :)